assets/js/blocks/cart-checkout/checkout/style.scss

```diff
@@ -224,6 +224,9 @@
 			.wc-block-components-checkbox {
 				clear: both;
 			}
+			& + .wc-block-components-text-input {
+				margin-top: 1.5em;
+			}
```

Comment:
Yeah, I think it might be easier to break it into its own pull, because the issue is a bit tricky. I could still reproduce it when using a country with different fields order:

![imatge](https://user-images.githubusercontent.com/3616980/89774894-01d1c480-db07-11ea-8da9-58bee0c55c9e.png)

I'm pretty sure I introduced it myself in #2877 ([relevant lines of code](https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/2877/files#diff-cf1bcc85aa4824861706c402d4d8e000R212-R215)). But that was required to avoid a huge margin between the billing address form and the order notes checkbox when there is no shipping neither payment methods.

So a fix for #2979 will need to be tested in all possible combinations (different countries which have different address fields + cart with/without shipping, payment, etc.).

Reply:
Pulled into its own PR https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/2989.